assets-sync.js

```diff
@@ -1407,15 +1407,15 @@ class WorldMigration {
                 return localPath;
             }
         } else {
-            const asset = this.assets.get(decodeURI(name));
+            const asset = this.assets.get(ForgeAssetSync.sanitizePath(decodeURIComponent(name)));
             const queryString = query ? `?${query}` : "";
             // Same path, not bazaar and same url.. so it's not coming from someone else's library
             if (asset && `${asset.url}${queryString}` === entityPath) {
-                return `${name}${queryString}`;
+                return `${asset.name}${queryString}`;
```

Comment:
We've already found the matching local asset. Reference it directly by its name and avoid further string manipulation.